src/dotnet/commands/dotnet-install/dotnet-install-tool/ProjectRestorer.cs

```diff
@@ -56,11 +56,11 @@ public ProjectRestorer(IReporter reporter = null)
             var command = new DotNetCommandFactory(alwaysRunOutOfProc: true)
                 .Create("restore", argsToPassToRestore);
 
-            if (_reporter != null)
+            if (verbosity == null || _reporter != null)
             {
                 command = command
-                    .OnOutputLine((line) => _reporter.WriteLine(line))
-                    .OnErrorLine((line) => _reporter.WriteLine(line));
+                    .OnOutputLine(line => WriteLine(_reporter ?? Reporter.Output, line, project))
```

Comment:
_reporter should not be null after constructor, if so could you fix it?

Reply:
`_reporter` can't default to a single value because output from `STDERR` should go to `Reporter.Error` and not `Reporter.Output`.  So we would need two fields.  I'd prefer one that says "direct all output to this reporter".

Actually, this just reminded me that your `tool-path` refactoring is now passing in `Reporter.Output` in the constructor where previously it was only being used for tests.  This breaks stderr output.

Reply:
Additionally, we'd need another flag field to indicate whether or not the original reporter passed in the constructor was null or not (if it's non-null then output must be redirected).  It's much simpler to allow the field to be null, imo.  But if you feel strongly about it, I could change to these fields:

```
        private readonly IReporter _reporter;
        private readonly IReporter _errorReporter;
        private readonly bool _forceOutputRedirection;
```

Reply:
I decided to fix it as describe.  I've pushed up the change.